model/item.cc

```diff
@@ -161,8 +161,14 @@ namespace minsky
  
   bool Item::visible() const 
   {
+	if (auto i=dynamic_cast<const IntOp*>(this))
+	  if (i->intVar->attachedToDefiningVar()) return false; 
+	if (auto o=operationCast()) 
+	  if (o->attachedToDefiningVar()) return false;
+	if (auto v=variableCast()) 
+	  if (v->attachedToDefiningVar()) return false;	  
```

Comment:
If attachedToDefiningVar() is virtual, which I think it is, then there is no need to do the *Cast() calls.

Reply:
Note pay attention to Switches, which have multiple outputs.